s4e-web/src/app/views/map-view/map-view.component.ts

```diff
@@ -222,6 +222,12 @@ export class MapViewComponent implements OnInit, OnDestroy {
     if (this.mapQuery.isLoading()) {
       return;
     }
+
+    // IMPORTANT!!! Prevent loading random date (1970-01-01) on first product visit
```

Comment:
It's not really random, it's arbitrary: https://en.wikipedia.org/wiki/Unix_time.

Reply:
You're right in the case of Unix time, but the comment is related to randomly showing the `NaN` value date. Maybe I wasn't too precised.